### PR TITLE
Electrical.Analog.Lines: addresses #3748 and #3755

### PR DESCRIPTION
**Pull request by AHaumer**
Included #3755 (thanks to @dietmarw !) since this is tidy up cosmetics when fixing #3748 (thanks to @HansOlsson for the suggestion of the optional ground). This is backwards compatible, no need for conversion.

### CODE REVIEW

**Review by christiankral — APPROVED** [review]
Looks correct

**Review by dietmarw — APPROVED** [review]
Looks good to me.

**Comment by beutlich** [comment]
Note, that I corrected the targeting milestone.

**Comment by christiankral** [comment]
@max-privato Do you have a proposal for the icon(s) so that we can proceed with is PR?

**Comment by AHaumer** [comment]
@max-privato Please provide your proposal for clarifying the documentation ...

**Comment by max-privato** [comment]
> @max-privato Do you have a proposal for the icon(s) so that we can proceed with is PR?

I thought this ticket was over. I'll have a look

**Comment by max-privato** [comment]
> @max-privato Please provide your proposal for clarifying the documentation ...

I forgot this ticket. I'll have a look

**Comment by max-privato** [comment]
I've worked a bit on the documentation and already built the main drawing
(below).

However I have some doubts on how the pin switch has been implemented:

1) The internal reference pin is called "ground" if disconnected from the
circuit ground; it is called "M" if it is an actual ground. This is very
confusing. I would call the internal pin "ref" (as in the picture below)
when independent, or "ground" if it is actually a Ground.
2) Consider the following code row:

> parameter Boolean useInternalGround=false "Default: false = internal
> ground / true = ground pin"

 The description and default seem to me wrong: consider that this is used
in the code:
>
>    Modelica.Electrical.Analog.Interfaces.NegativePin ground if not
> useInternalGround "Ground pin";
>    Modelica.Electrical.Analog.Basic.Ground M if useInternalGround;

the default should be changed to true and description improved. I would
change this row into:

> parameter Boolean useInternalGround=true "=true, the internal reference
> pin is to be a circuit ground"


Once it is agreed upon these two points, I could change the
documentation (and possibly the code) accordingly.

[image: Line_pict1.png]

Il giorno gio 11 mar 2021 alle ore 19:05 Anton Haumer <
***@***.***> ha scritto:

**Comment by christiankral** [comment]
> Once it is agreed upon these two points, I could change the
documentation (and possibly the code) accordingly.

> [image: Line_pict1.png]

Your image is unfortunately not included in this PR. Just paste it in your writing field...

**Comment by max-privato** [comment]
I've made all corrections according to my comment posted  four hours ago.
I've changed the code, the documentation and the two pictures.
The mo file and the two png's are attached here.
Despite all my efforts, I cannot be sure that everything is clear and smooth: any comments are welcome.

segment
segment_last
[M_Oline.zip](https://github.com/modelica/ModelicaStandardLibrary/files/6178090/M_Oline.zip)

**Comment by max-privato** [comment]
I've changed the model according to the concepts in my post dated 4 hours
ago.
I modified model, documentation, pictures. Everything is attached below.
Despite all my efforts, I cannot be sure that everything is ok. Comments
and corrections are welcome.
[image: segment.png]
[image: segment_last.png]



Il giorno dom 21 mar 2021 alle ore 16:39 Christian Kral <
***@***.***> ha scritto:

**Comment by dietmarw** [comment]
@AHaumer  You were a bit quick to merge since the images provided are not yet included in this PR. I probably should have blocked the PR but thought this not being necessary.

**Comment by dietmarw** [comment]
@max-privato I was trying to create a new PR on order to add the updates you submitted here and @AHaumer  missed to include before merging. The zip you attached is missing the updated images. Could you add those as well. It's fine to add them here as a comment. Best as the source Modelica from which you generated the images so I can create ones without canvas and grid which looks cleaner. Thanks.

**Comment by dietmarw** [comment]
@AHaumer I see that you did not take the other [comments](https://github.com/modelica/ModelicaStandardLibrary/pull/3756#issuecomment-803591198) into account. Was that on purpose or a simple oversight?

**Comment by dietmarw** [comment]
I've now reverted this since there are still some things that need to get resolved before it can finally get merged.

**Comment by beutlich** [comment]
> 
> 
> I've now reverted this since there are still some things that need to get resolved before it can finally get merged.

I labeled as invalid and changed milestone accordingly.

**Comment by AHaumer** [comment]
I've merged the PR as sson as we had a clean solution and two approvements.
For a long time there was no reaction from max-privato.

**Comment by dietmarw** [comment]
@AHaumer That is not true. @max-privato has provided the images you asked for and had also additional comments. You then simply merged the PR ignoring both. Those two things should be addressed. I can add the images but would like you to at least look at the [comments](https://github.com/modelica/ModelicaStandardLibrary/pull/3756#issuecomment-803591198)

**Comment by AHaumer** [comment]
You're right, but I expected max-privato to contribute to my PR instead of posting suggestions after I implemented a clean solution. 
Feel free to add the images ... Thanks!

**Comment by dietmarw** [comment]
@AHaumer  That is fine but could you give some feedback on the suggested improvements?

**Comment by max-privato** [comment]
> You're right, but I expected max-privato to contribute to my PR instead of posting suggestions after I implemented a clean solution.
> Feel free to add the images ... Thanks!

After posting comments **I changed the code and added the changed code in a zip file** (see may latest comment dated 21 March)

So, I gave my contribution to create a code that, for me, is well improved over the previous version, based on my comments.
Sorry for not creating a PR, but I'm not at ease with doing this and feared to do it badly. I hoped this did not bother anyone.
Since I was asked to post the images on this blog I did it. 

Today I'll look if I can find the original mo file from which I created the basic images, and the Visio file containing the images, added with further graphic elements (coloured curves and text). 

In case I succeed, I'll share them.

**Comment by max-privato** [comment]
In the enclosed zip you find:
- the revised mo file
- the two pictures
- two mo files from which the pictures are created

I confirm that the mo file is revised according to my comments and issued on March, 21, and pictures posted the same date. In the last message on that date I clearly explained that the code I'd put in the zip included modifications according to my previous comments.

The only tiny modification I made today was to change the name of the reference pin from "ref" to "refPin" (on March 21 it was "refPin" in the code, but "ref" in the picture)

I really hope these changes are included in of MSL.
Regards.

[M_Oline_full.zip](https://github.com/modelica/ModelicaStandardLibrary/files/6521110/M_Oline_full.zip)

**Comment by dietmarw** [comment]
Thanks. I will wait for @AHaumer's feedback on the suggested changes and then open up a new PR which will contain the final version including the updated images.

**Comment by AHaumer** [comment]
Well the suggestions seem to be ok, but it's hard to compare the versions. Since in my original PR cosmetic tidy-up (#3755) was included, I suggest that @dietmarw contributes the suggested changes to my PR / branch "Lines" and we'll review the result.

**Comment by max-privato** [comment]
Sorry for not having worked on the PR. I remember that @haumer you were so fast in implementing in your PR  a recommendation I issued on another ticket, that I had hardly started approaching mine!

However, In the code I sent to this ticket I and just added to AHaumer's version the changes discussed in this ticket, plus some minor clarity improvements.
So I suppose my implementation can be accepted, since it combines AHhamuer implementation and my comments. In case of need, I can give further explanations.

But I don't know how the actual process of managing MSL works, so I cannot do anything more than leave the room to you library officials. 

Good that finally we are finally getting to improve it!

**Comment by Haumer** [comment]
Seems like there is another A Haumer 👋😉

**Pull request merged**
